PyTorch/classification/train_classification.py

```diff
@@ -38,55 +50,82 @@ def train(dataloader, model, device, loss, learning_rate, momentum, weight_decay
             with profiler.profile(record_shapes=True, with_stack=True, profile_memory=True) as prof:
                 with profiler.record_function("model_inference"):
                     # Compute loss and perform backpropagation
-                    batch_loss = loss(model(X), y)
+                    if (model_str == 'inception_v3'):
+                        pred, _ = model(X)
+                    elif (model_str == 'googlenet'):
+                        pred, _, _ = model(X)
+                    else:
+                        pred = model(X)
+
+                    batch_loss = loss(pred, y)
                     batch_loss.backward()
 
                     if batch % optimize_after_batches == 0:
                         optimizer.step()
                         optimizer.zero_grad()
             print(prof.key_averages().table(sort_by="cpu_time_total", row_limit=1000))
-            break;
+            break
         else:
             # Compute loss and perform backpropagation
-            batch_loss = loss(model(X), y)
+            if (model_str == 'inception_v3'):
```

Comment:
same question as above - how would you handle new models that also have multiple return values?

Reply:
For new models and return more than 1 outputs, more "else if" conditions need to be added.